### PR TITLE
Placement of the `eval_time` argument

### PR DESCRIPTION
**Pull request by hfrick**
This PR moves the `eval_time` argument right after `metrics` (or `metric`) for user-facing functions:

- `fit_resamples()`, `tune_grid()`, `tune_bayes()`
- `last_fit()`
- `fit_best()`
- `show_best()`, `select_by_pct_loss()`
- `autoplot()` 

Corresponding changes should happen in finetune for `tune_race_anova()`, `tune_race_win_loss()`, and `tune_sim_anneal()`.

This PR does not touch `.estimate_metrics()` because the `eval_time` argument there should be removed, see #856. It also does not touch the argument order of the internal loop functions because the inconsistent naming of other arguements paired with passing arguments by position makes it likely to introduce bugs, so leaving this to #742.

The revdeps for these changes:
```
> revdepcheck::cloud_summary()
ℹ Syncing results to revdep/cloud.noindex/e7d0b88d-96f8-42a4-8b7e-a5ae07e69610
ℹ Comparing results
✔ agua 0.1.3                             ── E: 0     | W: 0     | N: 1    
✔ bonsai 0.2.1                           ── E: 0     | W: 0     | N: 0    
✔ finetune 1.1.0                         ── E: 1     | W: 0     | N: 0    
✔ finnts 0.4.0                           ── E: 0     | W: 0     | N: 0    
✔ healthyR.ai 0.0.13                     ── E: 1     | W: 1     | N: 0    
✔ healthyR.ts 0.3.0                      ── E: 0     | W: 0     | N: 1    
✔ HTRX 1.2.4                             ── E: 0     | W: 0     | N: 1    
✔ modeltime 1.2.8                        ── E: 0     | W: 0     | N: 1    
✔ modeltime.ensemble 1.0.3               ── E: 0     | W: 0     | N: 1    
✔ modeltime.resample 0.2.3               ── E: 0     | W: 0     | N: 1    
✔ nestedmodels 1.1.0                     ── E: 0     | W: 0     | N: 0    
✔ offsetreg 1.0.0                        ── E: 0     | W: 0     | N: 0    
✔ probably 1.0.3                         ── E: 0     | W: 0     | N: 0    
✔ shinymodels 0.1.1                      ── E: 0     | W: 0     | N: 0    
✔ stacks 1.0.3                           ── E: 0     | W: 0     | N: 0    
✔ tabnet 0.5.0                           ── E: 0     | W: 0     | N: 0    
✔ text 1.2.0                             ── E: 0     | W: 0     | N: 1    
✖ tidyclust 0.2.0                        ── E: 0  +1 | W: 0     | N: 0    
✔ tidydann 1.0.0                         ── E: 0     | W: 0     | N: 0    
✔ tidymodels 1.1.1                       ── E: 0     | W: 0     | N: 0    
✔ tidyposterior 1.0.1                    ── E: 1     | W: 0     | N: 0    
✖ tidysdm 0.9.3                          ── E: 0  +2 | W: 0  +1 | N: 0    
✔ timetk 2.9.0                           ── E: 0     | W: 0     | N: 1    
✔ usemodels 0.2.0                        ── E: 0     | W: 0     | N: 0    
✔ viralmodels 1.2.0                      ── E: 0     | W: 0     | N: 0    
✔ viruslearner 0.0.1                     ── E: 0     | W: 0     | N: 0    
✔ workflowsets 1.0.1                     ── E: 0     | W: 0     | N: 0   
```

### CODE REVIEW

File: R/last_fit.R [hunk 1]
```diff
@@ -141,30 +141,42 @@ last_fit.model_spec <- function(object, preprocessor, split, ..., metrics = NULL
...
-  last_fit_workflow(object, split, metrics, control, eval_time,
-                    add_validation_set)
+  last_fit_workflow(
+    object, 
+    split = split, 
+    metrics = metrics, 
+    eval_time = eval_time,
+    control = control, 
+    add_validation_set = add_validation_set
+  )
```
Comment (simonpcouch) [inline R/last_fit.R:171]: Thanks for adding the names here!

**Review by simonpcouch — COMMENTED** [review]
On board for mostly not touching `grid_code_paths.R`. Does `check_initial()` need a change here?

File: R/last_fit.R (hunk 1 above)
Comment (hfrick) [reply R/last_fit.R:171]: Safety! 😆

**Comment by hfrick** [comment]
Good shout! Added in [ff1af86](https://github.com/tidymodels/tune/pull/857/commits/ff1af868c50b39ba12217bd6305f2be63ffac2ef)

**Comment by hfrick** [comment]
The changes for finetune are in https://github.com/tidymodels/finetune/pull/104

The tests in extratests are being run for both PRs in https://github.com/tidymodels/extratests/pull/201

Without the changes to `check_initial()` they already ran successfully. The current run includes those changes + the finetune changes.

**Review by simonpcouch — APPROVED** [review]
Looking good. :)

From my perspective, this PR and it's finetune friend are ready to merge!

I'd say we wait to file revdep breakages until we've merged the coming dots changes and possible brier score introduction in #859, as there may be some overlap.

**Pull request merged**